go/store/datas/pull/puller.go

```diff
@@ -101,12 +101,25 @@ func NewPuller(
 		return nil, ErrIncompatibleSourceChunkStore
 	}
 
+	// walkAddrs can be used for getAddrs on the AddTableFile call as well.
+	getAddrs := func(c chunks.Chunk) chunks.GetAddrsCb {
+		return func(ctx context.Context, ins hash.HashSet, filter chunks.PendingRefExists) error {
+			return walkAddrs(c, func(h hash.Hash, _ bool) error {
+				if !filter(h) {
```

Comment:
Probably too difficult to change but this is the opposite of the usual filter idiom